### PR TITLE
#35 Add interactive text and website links for hotlines

### PR DESCRIPTION
**Pull request by lizzthabet**
@jaqarrick and I made progress on issue #35. This PR adds interactive icons for website and text for each hotline entry that has that information available. It doesn't include any work related to tags (which are also mentioned in #35).

*  Display the text contact information underneath each hotline entry that has text info
*  Updated `constants/PhoneNumbers` to store text information in an object with `content` and `number` properties, instead of a string
*  Add `IconGroup` component that's responsible for rendering phone, text, and website icons that link to the corresponding url

View from web app:
Screen Shot 2020-12-15 at 4 48 33 PM

View from iOS simulator:
Screen Shot 2020-12-15 at 4 48 00 PM

View from messages app on desktop after clicking sms link from web app:
Screen Shot 2020-12-15 at 4 49 30 PM

We tested in browser and Expo Android app without any issues. The iOS simulator had a couple bugs: phone number links (whose implementation didn't change in our code) threw an error; and sms texting links opened up in the messages app with the appropriate content, but not the appropriate phone number. After googling for a bit, our best guess is that both of these issues are due to limitations in the iOS simulator and not with the `tel` or `sms` links.

This is our first time contributing to this project, so feel free to let us know if there are any code style or implementation details that can be approved or better aligned with existing patterns!

A couple additional questions:
- [ ] Should the website url be listed as text, like the phone numbers, in addition to being linked with the website icon?
- [ ] Is there a better way to test on iOS devices?

### CODE REVIEW

File: constants/PhoneNumbers.ts [hunk 1]
```diff
@@ -209,6 +209,6 @@ export const PhoneNumbers: Props = {
       'Multilingual support',
     ],
     tel: '1-877-846-7369',
-    text: 'or text HOPENY',
+    text: {number: 'HOPENY'}, // TODO (JC): Make sure that this letter phone number format works properly
```
Comment (lizzthabet) [inline constants/PhoneNumbers.ts:212]: This comment can be removed!

File: constants/PhoneNumbers.ts [hunk 2]
```diff
@@ -81,14 +81,14 @@ export const PhoneNumbers: Props = {
...
-  //   text: 'Text GOT5 to 741741',
+  //   text: { content: 'GOT5', number: '741741' },
   // },
   {
     display: 'SAMHSA Disaster Distress National Helpline',
     hours: '24 Hour Access',
     tags: ['Post-Disaster', 'Crisis', 'Emotional Distress', 'Trauma', 'COVID-19', 'Unrest', 'Spanish'],
     tel: '1-800-985-5990',
-    text: 'or text TalkWithUs or Hablanos to 66746',
+    text: {content: 'TalkWithUs or Hablanos', number: '66746'},
```
Comment (rossPatton) [inline constants/PhoneNumbers.ts:91]: This text content works for the user reading, but when actually texting it should probably be either `TalkWithUs` or `Hablanos` depending on the user's language. But I think for right now, it's fine

**Review by rossPatton — APPROVED** [review]
looks good! tested it locally and things seem mostly fine to me. i've gone ahead and approved so merge anytime, but it would be nice to have that comment removed beforehand

to answer your questions:
- i think icon alone is probably fine for the website urls for now. but i don't think it would hurt to list the URL as plain text in the same size as the phone numbers. would probably need to be on it's own line
- regarding iOS testing, I don't know of another way besides testing on real devices. which  we've done, and I don't recall there being any issues. Regardless, unfortunately we're not on the app store so it's not anything we need to (currently) worry much about

**Comment by rossPatton** [comment]
gonna just go ahead and merge this!

**Comment by rossPatton** [comment]
just updating y'all - this has been rolled out to the play store!

**Comment by jaqarrick** [comment]
Hi Ross, Sorry about the delay in getting back to you - it's been a busy couple weeks for me! Thanks for merging for us. 

I'll keep an eye out for issues future issues!

**Pull request merged**